akka-actor/src/main/java/akka/dispatch/AbstractNodeQueue.java

```diff
@@ -155,12 +155,12 @@ public final T poll() {
     public final Node<T> pollNode() {
       final Node<T> tail = (Node<T>) Unsafe.instance.getObjectVolatile(this, tailOffset);
       Node<T> next = tail.next();
-      if (next == null && get() != tail) {
+      while (next == null && get() != tail && next != tail.next()) {
           // if tail != head this is not going to change until producer makes progress
           // we can avoid reading the head and just spin on next until it shows up
-          do {
-              next = tail.next();
-          } while (next == null);
+          // we're using Thread.yield to prevent use all available CPU
+          Thread.yield();
+          next = tail.next();
```

Comment:
```
     * A hint to the scheduler that the current thread is willing to yield
     * its current use of a processor. The scheduler is free to ignore this
     * hint.
```
I think it should be better to spin and swith to park sometime before start spin agin.

Reply:
@hepin1989 unfortunately but Thread.onSpinWait() that is in the fact just a pause if CPU and JVM supports it, but anyway it is available sine Java9 and until akka finish supports java8 Thread.yield() is the best option.

Reply:
Also if you read related issue you can understand that in near future I may close both an issue and the pull request as irrelevant